addons/stock/models/stock_quant.py

```diff
@@ -249,7 +249,7 @@ def check_product_id(self):
     def check_quantity(self):
         for quant in self:
             if float_compare(quant.quantity, 1, precision_rounding=quant.product_uom_id.rounding) > 0 and quant.lot_id and quant.product_id.tracking == 'serial':
-                raise ValidationError(_('A serial number should only be linked to a single product.'))
+                raise ValidationError(_('The serial number "%s" for product "%s" has already been assigned.' % (quant.lot_id.name, quant.product_id.name)))
```

Comment:
`quant.product_id.display_name`, use `display_name`, if there are variants it is better.

Reply:
You are right. Change applied. Thanks